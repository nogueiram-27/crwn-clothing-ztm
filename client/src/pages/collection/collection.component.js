import React from 'react';
import { connect } from 'react-redux';
import { selectCollection } from '../../redux/shop/shop.selectors';

import CollectionItem from '../../components/collection-item/collection-item.component';

import { CollectionPageContainer, TitleContainer, ItemsContainer } from './collection.styles'

const CollectionPage = ({ collection }) => {
    console.log('collection', collection);
    const { title, items } = collection;    
    return(
        <CollectionPageContainer>
            <TitleContainer>{title}</TitleContainer> 
            <ItemsContainer>
                {
                    items.map(item => (
                        <CollectionItem key={item.id} item={item} />
                    ))
                }
            </ItemsContainer>
        </CollectionPageContainer>
    )
}

const mapStateToProps = (state, ownProps) => {
    console.log('ownProps', ownProps)
    return{collection: selectCollection(ownProps.match.params.collectionId)(state)}
    
}

export default connect(mapStateToProps)(CollectionPage);