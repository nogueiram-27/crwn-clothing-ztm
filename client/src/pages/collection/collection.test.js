import React from 'react'
import { shallow } from 'enzyme'

import { CollectionPage } from './collection.component'
import CollectionItem from '../../components/collection-item/collection-item.component'; 

describe('CollectionPage', () => {
    let wrapper;
    let mockItems = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }]

    beforeEach(() => {
        const mockCollection = {
            items: mockItems,
            title: 'Test'
        }

        wrapper = shallow(<CollectionPage collection={ mockCollection } />)
    })

    it('should render CollectionPage component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render CollectionItem component as many times as mockItems size', () => {
        expect(wrapper.find(CollectionItem).length).toBe(mockItems.length)
    })

})