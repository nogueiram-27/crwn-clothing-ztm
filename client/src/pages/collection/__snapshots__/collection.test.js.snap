// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPage should render CollectionPage component 1`] = `ShallowWrapper {}`;
